To add command that lists backups enabled public networks

vldolot opened the pull request:
To add command that lists backups enabled public networks so that users can provide the correct combination of nodeType, clusterKey & networkSpecKey when creating a new node

New command:
`onf network-spec list-backups`

**NOTE:**
This requires the new `GET /api/v2/backup` API (from mx-api) to be deployed first before this command can be properly used.

Related JIRA ticket:
https://onfinality.atlassian.net/browse/OP-1309

Code changes:
```
modified:   cmd/networkspec/networkspec.go
- added listBackupsCmd()

modified:   pkg/service/networkspec.go
- added NetworkSpecBackups and Backups struct definitions
- added GetBackupNetworkSpecs() and GetBackups() functions
- added GetNodeTypeFromPruningMode() into service pkg as a method of Backups

modified:   sample/create-node.yaml
- modified hint comments to improve instructions on where to get the correct value combinations when using

added:       cmd/networkspec/listBackups.go
- added a new command to list all the networks that supports lightning restore/backups
```

Commit:
OP-1309

modified:   cmd/networkspec/networkspec.go
- added listBackupsCmd()

modified:   pkg/service/networkspec.go
- added NetworkSpecBackups and Backups struct definitions
- added GetBackupNetworkSpecs() and GetBackups() functions

modified:   sample/create-node.yaml
- modified hint comments to improve instructions on where to get the correct value combinations when using

added:       cmd/networkspec/listBackups.go
- added a new command to list all the networks that supports lightning restore/backups

vldolot commented on cmd/networkspec/listBackups.go:
```diff
@@ -0,0 +1,99 @@
+	return nsBackups
+}
+
+func getNodeTypeFromPruningMode(pruningMode string, protocol string) string {
```
Might be better to put this in a helper package?

454076513 replied:
In my opinion, Can be used as a `Backups ` method

vldolot replied:
Ok, I'll move it as a method in Backups. Thanks

454076513 commented on cmd/networkspec/listBackups.go:
```diff
@@ -0,0 +1,99 @@
+			}
+
+			// get backups supported public networks data
+			backups, err2 := service.GetBackups()
```
The err2 can use err above

vldolot replied:
Oh, right :) Still not used to var declarations in golang. I'll update the code. Thanks!

Commit:
OP-1309

modified:   cmd/networkspec/listBackups.go
- renamed err2 & err3 to err as suggested in comment

modified:   pkg/service/networkspec.go
- moved GetNodeTypeFromPruningMode() into service pkg as a method of Backups

Commit:
OP-1309

modified:   cmd/networkspec/listBackups.go
- added codes to convert returned string array from GetNodeTypeFromPruningModeAndProtocol() to string1

modified:   pkg/service/networkspec.go
- changed GetNodeTypeFromPruningMode() to GetNodeTypeFromPruningModeAndProtocol() and to return string array
